File: app/src/main/java/com/android/malsoftware/techadvisor/RecycleViewSelect/MySelectionPredicate.java
package com.android.malsoftware.techadvisor.RecycleViewSelect;

import androidx.annotation.NonNull;
import androidx.recyclerview.selection.SelectionTracker;

public class MySelectionPredicate extends SelectionTracker.SelectionPredicate<String> {

    private Object mLastKey = null;

    @Override
    public boolean canSetStateForKey(@NonNull String key, boolean nextState) {
        /*if (mLastKey == key) return false;
        mLastKey = key;*/
        return true;
    }

    @Override
    public boolean canSetStateAtPosition(int position, boolean nextState) {
        return false;
    }

    @Override
    public boolean canSelectMultiple() {
        return false;
    }
}

File: app/src/main/java/com/android/malsoftware/techadvisor/RecycleViewSelect/MySelectionPredicate.kt
package com.android.malsoftware.techadvisor.RecycleViewSelect

import androidx.recyclerview.selection.SelectionTracker

class MySelectionPredicate : SelectionTracker.SelectionPredicate<String>() {

    private val mLastKey: Any? = null

    override fun canSetStateForKey(key: String, nextState: Boolean): Boolean {
        /*if (mLastKey == key) return false;
        mLastKey = key;*/
        return true
    }

    override fun canSetStateAtPosition(position: Int, nextState: Boolean): Boolean {
        return false
    }

    override fun canSelectMultiple(): Boolean {
        return false
    }
}